ui/src/views/Query.vue

```diff
@@ -82,32 +64,40 @@ export default {
   data() {
     return {
       queryResultsIsLoading: true,
+      workloadUUID: null,
+      projectName: null,
       showResults: {
         results: "You might need to login to query for workload!"
       },
       valid: true,
-      dates: ["2020-01-10", "2020-01-12"],
-      environment: [{ text: "gotc-dev" }],
-      environmentRules: [v => !!v || "Environment is required"],
-      environmentSelected: ""
+      rules: {
+        workloadUUID: v => {
+          const pattern = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/i
```

Comment:
Wild. Where does this come from?

Reply:
😅  Got it and modified from https://github.com/microsoft/uuid-validate/blob/06554db1b093aa6bb429156fa8964e1cde2b750c/index.js#L2, I guess that's following https://tools.ietf.org/html/rfc4122#section-4.1.7. will add a reference.